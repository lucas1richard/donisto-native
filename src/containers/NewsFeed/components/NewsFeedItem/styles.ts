import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  wrapper: {
    marginTop: 20,
    borderRadius: 10,
    elevation: 3,
    shadowColor: '#555',
    shadowOpacity: 0.20,
    shadowOffset: {
      width: 2,
      height: 4
    },
    shadowRadius: 6,
    backgroundColor: '#fff'
  }
});

export default styles;
