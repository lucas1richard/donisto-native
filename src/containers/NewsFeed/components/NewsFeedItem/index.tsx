import * as React from 'react';
import { View } from 'react-native';
import styles from './styles';
import TopRow from './components/TopRow';
import BottomRow from './components/BottomRow';
import Img from './components/Img';

export interface INewsFeedItemProps {
  item: any;
  img: any;
  navDetail: () => any
}

export interface INewsFeedItemState {
  modalVisible: boolean;
}

class NewsFeedItem extends React.Component<INewsFeedItemProps, INewsFeedItemState> {
  render() {
    const { item, img, navDetail } = this.props;
    return (
      <View>
        <View style={styles.wrapper}>
          <TopRow name={item.organization.name} />
          <Img
            navDetail={navDetail}
            img={img}
          />
          <BottomRow />
        </View>
      </View>
    );
  }
}

export default NewsFeedItem;
