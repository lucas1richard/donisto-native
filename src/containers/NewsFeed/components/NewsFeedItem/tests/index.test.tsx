import * as React from 'react';
import { shallow } from 'enzyme';
import NewsFeedItem, { INewsFeedItemProps } from '..';

describe('<NewsFeedItem />', () => {
  const renderComponent = (props: INewsFeedItemProps) => shallow(<NewsFeedItem {...props} />);
  it('renders', () => {
    expect(renderComponent({
      item: {
        organization: {
          name: 'mockOrgName'
        }
      },
      img: 'mockImage',
      navDetail: jest.fn()
    })).toMatchSnapshot();
  });
});