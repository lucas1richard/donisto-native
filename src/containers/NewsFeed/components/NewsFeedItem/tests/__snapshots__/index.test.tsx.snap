// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsFeedItem /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsFeedItem
    img="mockImage"
    item={
      Object {
        "organization": Object {
          "name": "mockOrgName",
        },
      }
    }
    navDetail={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "elevation": 3,
            "marginTop": 20,
            "shadowColor": "#555",
            "shadowOffset": Object {
              "height": 4,
              "width": 2,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 6,
          }
        }
      >
        <TopRow
          name="mockOrgName"
        />
        <Img
          img="mockImage"
          navDetail={[MockFunction]}
        />
        <BottomRow />
      </Component>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TopRow
            name="mockOrgName"
          />,
          <Img
            img="mockImage"
            navDetail={[MockFunction]}
          />,
          <BottomRow />,
        ],
        "style": Object {
          "backgroundColor": "#fff",
          "borderRadius": 10,
          "elevation": 3,
          "marginTop": 20,
          "shadowColor": "#555",
          "shadowOffset": Object {
            "height": 4,
            "width": 2,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 6,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "mockOrgName",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "img": "mockImage",
            "navDetail": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "elevation": 3,
              "marginTop": 20,
              "shadowColor": "#555",
              "shadowOffset": Object {
                "height": 4,
                "width": 2,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 6,
            }
          }
        >
          <TopRow
            name="mockOrgName"
          />
          <Img
            img="mockImage"
            navDetail={[MockFunction]}
          />
          <BottomRow />
        </Component>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TopRow
              name="mockOrgName"
            />,
            <Img
              img="mockImage"
              navDetail={[MockFunction]}
            />,
            <BottomRow />,
          ],
          "style": Object {
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "elevation": 3,
            "marginTop": 20,
            "shadowColor": "#555",
            "shadowOffset": Object {
              "height": 4,
              "width": 2,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 6,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "mockOrgName",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "img": "mockImage",
              "navDetail": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
